internal/store/selector/selector.go

```diff
@@ -0,0 +1,256 @@
+package selector
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+	"sync"
+
+	"github.com/flightctl/flightctl/internal/flterrors"
+	gormschema "gorm.io/gorm/schema"
+)
+
+var (
+	cacheStore           = &sync.Map{}
```

Comment:
nit: are there any drawbacks to having the cache as a global?

Reply:
Gorm uses quite a similar approach :
https://github.com/go-gorm/gorm/blob/deceebfab8c460cfee229233aded2821ac6b08eb/gorm.go#L173

This is to save some reflection processing for the same model fields.